src/Impostor.Server/Net/Client.cs

```diff
@@ -348,7 +348,9 @@ public override async ValueTask HandleDisconnectAsync(string reason)
             {
                 if (Player != null)
                 {
-                    await Player.Game.HandleRemovePlayer(Id, DisconnectReason.ExitGame);
+                    // The client never sends over the real disconnect reason so we always assume ExitGame
+                    var isRemote = reason == "The remote sent a disconnect request";
```

Comment:
Magic string, consider moving it into a special string in Hazel that we can import from there in case the message is changed in Hazel.

Optional as we forked Hazel.

Reply:
The real solution would be adding an `IsInternal` or `IsRemote` boolean property to `DisconnectedEventArgs`, but I don't think it's worth the effort right now